xcodeproj/internal/xcode_targets.bzl

```diff
@@ -261,45 +261,59 @@ def _to_xcode_target_product(product):
         _additional_files = product.framework_files,
     )
 
-def _merge_xcode_target(*, src, dest):
-    """Creates a new `xcode_target` by merging the values of `src` into `dest`.
+def _merge_xcode_targets(*, srcs, dest):
```

Comment:
Based on my other comments, I would pass in two arguments here: `swift_src` and `non_swift_src`. We can make it clear that we can merge up to 2 targets, and grab specific things from each one.